js/src/base-component.js

```diff
@@ -38,6 +38,12 @@ class BaseComponent {
     return Data.get(element, this.DATA_KEY)
   }
 
+  static getInstanceOrNew(element, config = {}) {
```

Comment:
Better name => `getOrCreateInstance` :thinking:

Reply:
I am not sure which is best, but it is very easy to rename, if we want it